Modules/_io/winconsoleio.c

```diff
@@ -265,7 +263,10 @@ _io__WindowsConsoleIO___init___impl(winconsoleio *self, PyObject *nameobj,
     int fd_is_own = 0;
     HANDLE handle = NULL;
 
-    assert(PyObject_TypeCheck(self, (PyTypeObject *)&PyWindowsConsoleIO_Type));
+#ifdef Py_DEBUG
```

Comment:
You should use ``#ifndef NDEBUG`` instead.

You can build Python in release mode with assertions: https://docs.python.org/dev/using/configure.html#cmdoption-with-assertions

Reply:
Thanks, I will fix it in a follow-up PR (there will be many before we're done with _io!)